Code/Mantid/Framework/DataHandling/src/LoadDetectorsGroupingFile.cpp

```diff
@@ -384,9 +387,12 @@ void LoadDetectorsGroupingFile::generateNoInstrumentGroupWorkspace() {
 /*
  * Initialization
  */
-LoadGroupXMLFile::LoadGroupXMLFile() {
-  mStartGroupID = 1;
-  return;
+LoadGroupXMLFile::LoadGroupXMLFile(): m_instrumentName(""),
+    m_userGiveInstrument(false), m_date(""), m_userGiveDate(false),
+    m_description(""), m_userGiveDescription(false),
+    m_pDoc(NULL), m_pRootElem(NULL),  m_groupComponentsMap(),
+    m_groupDetectorsMap(), m_groupSpectraMap(), m_startGroupID(1),
+    m_groupNamesMap() {
```

Comment:
You're right, but I preferred to explicitly include all the members in all cases. It looks like many issues with constructor initializations started when someone added a new member and forgot to initialize it in the constructor. When the list is not exhaustive you can easily step over some members that need initialization without noticing. I'm hoping that having an exhaustive list will help keeping all the data members there, including the ones that really need explicit initialization.